File: AD-HOC/3058.py
import math

n = int(input())
valores = []

for i in range(n):
  entrada = list(map(float, input().split()))
  preco = entrada[0]
  gramas = entrada[1]
  total = (preco/gramas)*1000
  valores.append(total)

print("{:.2f}".format(min(valores)))

File: AD-HOC/3058.js
let n = Number(lines.shift());
let valores = [];

for (let i = 0; i < n; i++) {
  let [preco, gramas] = lines[i].split(' ').map(Number);
  let total = (preco/gramas)*1000;
  valores.push(total);
}

console.log(Math.min(...valores).toFixed(2));
